### PR TITLE
Review followup for #8036

### PR DESCRIPTION
**Pull request by kripken**
Main fix here is that regex was wrong - it needs to parse hexadecimal digits.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2208,7 +2208,7 @@ def parse_symbols(output, include_internal=False):
         continue # e.g.  filename.o:  , saying which file it's from
       parts = [seg for seg in line.split(' ') if len(seg)]
       # pnacl-nm will print zero offsets for bitcode, and newer llvm-nm will print present symbols as  -------- T name
-      if len(parts) == 3 and parts[0] == "--------" or re.match(r'^\d\d\d\d\d\d\d\d$', parts[0]):
+      if len(parts) == 3 and parts[0] == "--------" or re.match(r'^[\dabcdef][\dabcdef][\dabcdef][\dabcdef][\dabcdef][\dabcdef][\dabcdef][\dabcdef]$', parts[0]):
```
Comment (sbc100) [inline tools/shared.py]: Why this?  

Can you do `[\da-f]` instead?

File: tests/test_other.py [hunk 2]
```diff
@@ -225,7 +225,11 @@ def test_emcc_2(self):
           print('(no output)')
           continue
         syms = Building.llvm_nm(target)
-        assert len(syms.defs) == 1 and ('main' in syms.defs or '__original_main' in syms.defs), 'Failed to generate valid bitcode'
+        assert 'main' in syms.defs
+        if self.is_wasm_backend():
+          assert len(syms.defs) == 2 and '__original_main' in syms.defs
+        else:
+          assert len(syms.defs) == 1
```
Comment (sbc100) [inline tests/test_other.py]: The name `__original_main` seems like a detail we probably don't want to care about here.

How about just "assert len(syms.defs) <= 2"?

File: tools/shared.py (hunk 1 above)
Comment (jgravelle-google) [reply tools/shared.py]: also: `[\da-f]{8}$`?

File: tests/test_other.py (hunk 2 above)
Comment (kripken) [reply tests/test_other.py]: I agree the name is too detailed, removing. I think it's useful to keep `== 2` in this path though, so we are alerted if this changes.

File: tools/shared.py (hunk 1 above)
Comment (kripken) [reply tools/shared.py]: thanks, yeah, that's nicer.

**Pull request merged**